src/NewPost.js

```diff
@@ -0,0 +1,114 @@
+import React, { Component } from "react";
+import { data } from "./data";
+
+export class NewPost extends Component {
+  state = {
+    id: 0,
+    title: "",
+    author: "",
+    createdOn: "",
+    body: "",
+    tags: ""
+  };
+
+  createDateOnSubmit = () => {
+    const date = new Date();
+    let year = date.getFullYear();
+    let month = date.getMonth();
+    let day = date.getDate();
+    const dateNow = `${year}-${month}-${day}`;
+    return dateNow;
+  };
+
+  createUniqueIdOnSubmit = () => {
+    const id = data.length;
+    return id;
+  };
+
+  convertTagsToArray = tags => {
+    let resultArray = [];
+
+    if (tags != null) {
+      const splitOn = ",";
+
+      resultArray = tags.split(splitOn);
+      resultArray = resultArray.map(function(str) {
+        return str.trim();
+      });
+    }
+
+    return resultArray;
+  };
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    });
+  };
+
+  handleSubmit = e => {
+    e.preventDefault();
+
+    const resultArray = this.convertTagsToArray(this.state.tags);
+    let blogPost = Object.assign({}, this.state);
+    blogPost.tags = resultArray;
+
+    const newId = this.createUniqueIdOnSubmit(this.state.id);
```

Comment:
seems like there's no need to keep `id` in state in the first place. `createUniqueIdOnSubmit` doesn't use it

Reply:
I removed id from state but then id does not show up as a property in blogPost in console.

Reply:
upon further inspection...id does show up in blogPost Object.